source/Mlos.Python/mlos/OptimizerEvaluationTools/unit_tests/TestOptimizerEvaluator.py

```diff
@@ -0,0 +1,218 @@
+        final_optimizer_from_report = pickle.loads(optimizer_evaluation_report.pickled_optimizers_over_time[99])
+
+        for _ in range(100):
+            self.assertTrue(final_optimizer_from_disk.suggest() == final_optimizer_from_report.suggest())
+
+
+    def test_named_configs(self):
+        """Tests named optimizer configurations against named objective functions.
+
+        It is prohibitively expensive to test the entire cross product so we test only its subset, but in such a way that
+        each configuration will be tested at least once.
+        """
+        optimizer_named_configs = bayesian_optimizer_config_store.list_named_configs()
+        num_optimizer_configs = len(optimizer_named_configs)
+        objective_function_named_configs = objective_function_config_store.list_named_configs()
+        num_objective_function_configs = len(objective_function_named_configs)
+
+        num_tests = max(num_optimizer_configs, num_objective_function_configs)
+
+        with traced(scope_name="parallel_tests"), concurrent.futures.ProcessPoolExecutor(max_workers=4) as executor:
+            outstanding_futures = set()
+
+            for i in range(num_tests):
+                named_optimizer_config = optimizer_named_configs[i % num_optimizer_configs]
+                named_objective_function_config = objective_function_named_configs[i % num_objective_function_configs]
+
+                print("#####################################################################################################")
+                print(named_optimizer_config)
+                print(named_objective_function_config)
+
+                optimizer_evaluator_config = optimizer_evaluator_config_store.get_config_by_name(name="parallel_unit_tests_config")
+                optimizer_config = named_optimizer_config.config_point
+                objective_function_config = named_objective_function_config.config_point
+
+                optimizer_evaluator = OptimizerEvaluator(
+                    optimizer_evaluator_config=optimizer_evaluator_config,
+                    objective_function_config=objective_function_config,
+                    optimizer_config=optimizer_config
+                )
+
+                future = executor.submit(optimizer_evaluator.evaluate_optimizer)
+                outstanding_futures.add(future)
+
+            done_futures, outstanding_futures = concurrent.futures.wait(outstanding_futures, return_when=concurrent.futures.ALL_COMPLETED)
+
+            for future in done_futures:
+                optimizer_evaluation_report = future.result()
+                self.assertTrue(optimizer_evaluation_report.success)
+                mlos.global_values.tracer.trace_events.extend(optimizer_evaluation_report.execution_trace)
+
+                with pd.option_context('display.max_columns', 100):
+                    print(optimizer_evaluation_report.regression_model_goodness_of_fit_state.get_goodness_of_fit_dataframe(DataSetType.TRAIN).tail())

```

Comment:
Is the goal to manually check the output? Or why are we printing in the unit test?

Reply:
Of course not :) But I find the output useful during development.